packages/router/__tests__/router-test.ts

```diff
@@ -1957,19 +1950,40 @@ describe("a router", () => {
       });
       expect(shouldRevalidate.mock.calls.length).toBe(0);
 
-      // Normal navigations should not trigger fetcher revalidations
+      // Normal navigations should trigger fetcher shouldRevalidate with
+      // defaultShouldRevalidate=false
```

Comment:
This test used to incorrectly assert that we did not even call `shouldRevalidate`.  But instead, on normal navigations we should be calling `shouldRevalidate` on fetcher routes if it exists, and giving it a `defaultShouldRevalidate:false` value since it's just a normal navigation.  This aligns with our goal of giving the user op-in and opt-out control